src/common/packager.ts

```diff
@@ -36,50 +36,54 @@ export class Packager {
-                        // The packager will continue running while we debug the application, so we can"t
-                        // wait for this command to finish
-
-                        let spawnOptions = { env: childEnvForDebugging };
-
-                        new CommandExecutor(this.projectPath).spawnReactPackager(args, spawnOptions, outputChannel).then((packagerProcess) => {
-                            this.packagerProcess = packagerProcess;
-                            executedStartPackagerCmd = true;
-                        });
-                    }).done();
-            }
-        });
+        return this.isRunning()
+            .then(running => {
+                if (!running) {
+                    return this.monkeyPatchOpnForRNPackager()
+                        .then(() => {
+                            let args = ["--port", Packager.PORT];
+                            let childEnvForDebugging = Object.assign({}, process.env, { REACT_DEBUGGER: "echo A debugger is not needed: " });
+
+                            Log.logMessage("Starting Packager", outputChannel);
+                            // The packager will continue running while we debug the application, so we can"t
+                            // wait for this command to finish
+
+                            let spawnOptions = { env: childEnvForDebugging };
 
-        return this.awaitStart().then(() => {
-            if (executedStartPackagerCmd) {
-                Log.logMessage("Packager started.", outputChannel);
-            } else {
-                Log.logMessage("Packager is already running.", outputChannel);
-                if (!outputChannel) {
-                    Log.logMessage("Warning: Debugging is not supported if the React Native Packager is not started within VS Code. If debugging fails, please kill other active React Native packager processes and retry.", outputChannel);
+                            // TODO - PROMISE: We need to consume the result of this spawn
+                            new CommandExecutor(this.projectPath).spawnReactPackager(args, spawnOptions, outputChannel).then((packagerProcess) => {
+                                this.packagerProcess = packagerProcess;
+                                executedStartPackagerCmd = true;
+                            });
+                        });
+                }
+            })
+            .then(() =>
+                this.awaitStart())
+            .then(() => {
+                if (executedStartPackagerCmd) {
+                    Log.logMessage("Packager started.", outputChannel);
+                } else {
+                    Log.logMessage("Packager is already running.", outputChannel);
+                    if (!outputChannel) {
+                        Log.logMessage("Warning: Debugging is not supported if the React Native Packager is not started within VS Code. If debugging fails, please kill other active React Native packager processes and retry.", outputChannel);
```

Comment:
Changed to Log.logWarning.
You are correct, it's potentially warning erroneously but at the moment we don't know from the debugee process who launched the packager. I'm going to add a TODO so we can review this after the updates we are making to the packager, so we can make sure that we stop printing this if it's not necessary.
                        // TODO: This warning is printted incorrectly when the packager was started from the command palette. Fix it.